### ISSUE TITLE
Unable to parse my libraryfolders.vdf made by Steam on Windows

### ISSUE THREAD

**Issue by Enduriel** [WilliamVenner/steamlocate-rs#2]
So I'm not sure what's going on here but for some reason my specific libraryfolders.vdf, I've been trying to figure it out and it's something to do with the very last value: "E:\\Steam" being recognized as a } and the parser therefore erroring out. I've been unable to figure it out for hours so any help would be much appreciated.

[Anonymized libraryfolders](https://github.com/WilliamVenner/steamlocate-rs/files/6515811/Mylibraryfolders.zip)

I've also raised this issue on the steamy-vdf crate github but it seems to be dead unfortunately :/

**Comment by WilliamVenner** [comment]
I checked my libraryfolders.vdf file, which is roughly this:

```json
"LibraryFolders"
{
	"TimeNextStatsReport"		"0000002"
	"ContentStatsID"		"-123456789"
	"1"		"F:\\Steam"
	"2"		"E:\\Steam"
	"3"		"D:\\Steam"
}
```

Which seems to work for me.

**Comment by LovecraftianHorror** [comment]
I was able to recreate the issue with the provided file with the following code

`Mylibraryfolders.vdf`

```text
"LibraryFolders"
{
	"TimeNextStatsReport"		"0000002"
	"ContentStatsID"		"-123456789"
	"1"		"D:\\SteamLibrary"
	"2"		"E:\\Steam"
}
```

```rust
fn main() {
    let _ = steamy_vdf::load("Mylibraryfolders.vdf").unwrap();
}

```

Gets me

```text
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Parse', src/main.rs:2:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I really don't know what specifically is causing a parse error though. I had other problems with using `steamy-vdf` and `vdf-serde` so I ended up releasing my own [parsing](https://lib.rs/crates/keyvalues-parser) and [serde](https://lib.rs/crates/keyvalues-serde) libraries instead. I just released the first versions of both so there are likely several things that can be improved, but there's a decent number of unit tests for everything, a fuzzing suite, and I'm open to new changes and contributions.

I guess it's worth noting that the keyvalues-parser handles both of the example libraryfolders.vdf without issue. Here's the output from the problematic one

```rust
use keyvalues_parser::Vdf;
use std::fs;

fn main() {
    let vdf_text = fs::read_to_string("Mylibraryfolders.vdf").unwrap();
    let vdf = Vdf::parse(&vdf_text).unwrap();
    println!("{:#?}", vdf);
}
```

```text
Vdf {
    key: "LibraryFolders",
    value: Obj(
        {
            "1": [
                Str(
                    "D:\\SteamLibrary",
                ),
            ],
            "2": [
                Str(
                    "E:\\Steam",
                ),
            ],
            "ContentStatsID": [
                Str(
                    "-123456789",
                ),
            ],
            "TimeNextStatsReport": [
                Str(
                    "0000002",
                ),
            ],
        },
    ),
}
```

There are quite a few differences from `steamy-vdf` though, just as a heads up.

**Comment by Dudemanjude** [comment]
It seems like this issue may have been fixed by pull request #4 (I haven't done any testing, I've just been looking at this library for one of my projects), so shouldn't this be closed now?

**Comment by WilliamVenner** [comment]
Brilliant! Thank you to @LovecraftianHorror
